great_expectations/core/__init__.py

```diff
@@ -329,9 +329,9 @@ def __init__(self, run_name=None, run_time=None):
                     f'Unable to parse provided run_time str ("{run_time}") to datetime. Defaulting '
                     f"run_time to current time."
                 )
-                run_time = datetime.datetime.now(datetime.timezone.utc)
+                run_time = datetime.datetime.utcnow()
 
-        self._run_time = run_time or datetime.datetime.now(datetime.timezone.utc)
+        self._run_time = run_time or datetime.datetime.utcnow()
```

Comment:
This doesn't quite seem like we're assuming the run_time is in UTC, but rather later formatting it as if it is UTC whether it is or isn't. I think instead we should check at this point for non-utc tz-aware datetimes and convert them to UTC.